### PR TITLE
Release v1.2.0

### PR DESCRIPTION
**Pull request by jathanism**
Request to merge `develop` into `main` for release v1.2.0!

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -25,34 +25,36 @@ HEALTHCHECK --interval=5s --timeout=5s --start-period=5s --retries=1 CMD curl --
...
 RUN curl -Lo /usr/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.0.0/hadolint-Linux-x86_64 && \
     chmod +x /usr/bin/hadolint
 
 # Install Poetry manually from GitHub because otherwise it installs its own
 # dependencies globally which may conflict with ours.
 # https://python-poetry.org/docs/#osx-linux-bashonwindows-install-instructions
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py -o /tmp/get-poetry.py && \
-    python /tmp/get-poetry.py && \
-    rm -f /tmp/get-poetry.py
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py -o /tmp/install-poetry.py && \
```
Comment (glennmatthews) [inline docker/Dockerfile:47]: I didn't notice we'd made this change already. This fixes #918.

**Review by glennmatthews — APPROVED** [review]
Just to confirm, I think we should ideally merge the following PRs before cutting this release:

- [x] #1007 
- [x] #1041
- [x] #1083
- [x] #1105
- [x] #1126
- [x] #1142 
- [x] #1143
- [x] #1149
- [x] #1160

File: mkdocs.yml [hunk 2]
```diff
@@ -75,6 +75,7 @@ nav:
         - Custom Fields: 'additional-features/custom-fields.md'
         - Custom Links: 'models/extras/customlink.md'
         - Export Templates: 'models/extras/exporttemplate.md'
+        - Filters: 'additional-features/filters.md'
```
Comment (glennmatthews) [inline mkdocs.yml:78]: Wonder if we should elaborate on this as `Template Filters` or similar? To make it less ambiguous with concepts like queryset filters?

File: nautobot/docs/additional-features/computed-fields.md [hunk 3]
```diff
@@ -37,13 +37,13 @@ Computed field templates can utilize the context of the object the field is bein
 
 ## Computed Field Template Filters
 
-Computed field templates can also utilize custom Jinja2 filters that have been registered via plugins. These filters can be used by providing the name of the filter function. As an example:
+Computed field templates can also utilize built-in Jinja2 filters or custom one that have been registered via plugins. These filters can be used by providing the name of the filter function. As an example:
```
Comment (glennmatthews) [inline nautobot/docs/additional-features/computed-fields.md:40]:
```suggestion
Computed field templates can also utilize built-in Jinja2 filters or custom ones that have been registered via plugins. These filters can be used by providing the name of the filter function. As an example:
```

File: nautobot/docs/additional-features/custom-fields.md [hunk 4]
```diff
@@ -38,11 +38,14 @@ Nautobot supports limited custom validation for custom field values. Following a
 * Text: Regular expression (optional)
 * Integer: Minimum and/or maximum value (optional)
 * Selection: Must exactly match one of the prescribed choices
+  * Selection Fields: Regular expression (optional)
 
 ### Custom Selection Fields
 
 Choices are stored as independent values and are assigned a numeric weight which affects their ordering in selection lists and dropdowns. Note that choice values are saved exactly as they appear, so it's best to avoid superfluous punctuation or symbols where possible.
 
+Regular expression can optionally be defined on custom selection choices to validate the field choice in the user-interface and the API.
```
Comment (glennmatthews) [inline nautobot/docs/additional-features/custom-fields.md:47]:
```suggestion
A regular expression can optionally be defined on custom selection choices to validate the defined field choices in the user interface and the API.
```

File: nautobot/docs/additional-features/filters.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
+# Additional Filters
```
Comment (glennmatthews) [inline nautobot/docs/additional-features/filters.md:1]: Perhaps?
```suggestion
# Additional Template Filters
```

File: nautobot/docs/additional-features/filters.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+## Introduction
+
+Nautobot uses 2 template engines internally, Django Template and Jinja2.
+Django Template is used to render the UI pages and Jinja2 is used for all extensibility features like: computed fields, custom links etc ...
```
Comment (glennmatthews) [inline nautobot/docs/additional-features/filters.md:6]:
```suggestion
Django Template is used to render the UI pages and Jinja2 is used for features such as computed fields, custom links, export templates, etc.
```

File: nautobot/docs/configuration/authentication/sso.md [hunk 7]
```diff
@@ -42,10 +42,10 @@ Install the system dependencies as `root`:
 $ sudo apt install -y libxmlsec1-dev libxmlsec1-openssl pkg-config
 ```
 
-Install the `saml` Python extra as the`nautobot` user.
+Install the `sso` Python extra as the`nautobot` user.
```
Comment (glennmatthews) [inline nautobot/docs/configuration/authentication/sso.md:45]:
```suggestion
Install the `sso` Python extra as the `nautobot` user.
```

File: nautobot/docs/release-notes/version-1.2.md [hunk 8]
```diff
@@ -116,12 +124,67 @@ Nautobot core applications and plugins can now both define panels, groups, and i
...
+!!! note
+    Because `JobLogEntry` records reference their associated `JobResult`, the pattern `job.job_result = JobResult()` (creating only an in-memory `JobResult` object, rather than a database entry) will no longer work. Instead you will need to create a proper JobResult database object `job.job_result = JobResult.objects.create(...)`.
+
 #### Slug fields are now Optional in CSV import, REST API and ORM ([#493](https://github.com/nautobot/nautobot/issues/493))
 
 All models that have `slug` fields now use `AutoSlugField` from the `django-extensions` package. This means that when creating a record via the REST API, CSV import, or direct ORM Python calls, the `slug` field is now fully optional; if unspecified, it will be automatically assigned a unique value, just as how a `slug` is auto-populated in the UI when creating a new record.
 
 Just as with the UI, the `slug` can still always be explicitly set if desired.
 
+## v1.2.0 (2021-12-14)
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-1.2.md:143]:
```suggestion
## v1.2.0 (2021-12-15)
```

File: nautobot/docs/rest-api/filtering.md [hunk 9]
```diff
@@ -85,3 +85,32 @@ String based (char) fields (Name, Address, etc) support these lookup expressions
...
+- `net_contains_or_equals` - Given a network, determine which networks contain or is the provided network e.g. `network__net_contains="192.168.0.0/16"` would include 192.0.0.0/8 and 192.168.0.0/16 in the result
+- `net_equals` - Given a network, determine which which networks are an exact match. e.g. `network__net_equals="192.168.0.0/16"` would include only 192.168.0.0/16 in the result
+- `net_host` - Determine which networks are parent of the provided IP, e.g. `host__net_host="10.0.0.1"`  would include 10.0.0.1/32 and 10.0.0.0/24 in the result
+- `net_host_contained` - Given a network, select IPs whose host address (regardless of its subnet mask) falls within that network , e.g. `host__net_host_contained="10.0.0.0/24"` would include hosts 10.0.0.1/8 and 10.0.0.254/32 in the result
+- `net_in` - Given a list of networks, select addresses (regardless of their subnet masks) within those networks, e.g. `host__net_in=["10.0.0.0/24", "2001:db8::/64"]` would include hosts 10.0.0.1/16 and 2001:db8::1/65 in the result
+- `family` - Given an IP address family of 4 or 6, provide hosts or networks that are that IP version type, e.g. `host__family=6` would include 2001:db8::1 in the result
```
Comment (glennmatthews) [inline nautobot/docs/rest-api/filtering.md:114]: Might be nice to get #1159 merged before release?

File: nautobot/utilities/templatetags/helpers.py [hunk 10]
```diff
@@ -212,49 +320,102 @@ def has_one_or_more_perms(user, permissions_list):
...
 
-@register.filter
+@library.filter()
+@register.filter()
 def settings_or_config(key):
+    """Get a value from Django settings (if specified there) or Constance configuration (otherwise)."""
     return get_settings_or_config(key)
 
 
+@register.filter
```
Comment (glennmatthews) [inline nautobot/utilities/templatetags/helpers.py:401]:
```suggestion
@library.filter()
@register.filter()
```

We also need to add a couple unit test cases for this filter now that we have tests for the other ones in this file. I'm fine with deferring that for now though.

**Review by glennmatthews — APPROVED** [review]

File: mkdocs.yml (hunk 2 above)
Comment (jathanism) [reply mkdocs.yml:78]: Yes, I think we should.

File: nautobot/docs/rest-api/filtering.md (hunk 9 above)
Comment (jathanism) [reply nautobot/docs/rest-api/filtering.md:114]: Merged.

**Pull request merged**